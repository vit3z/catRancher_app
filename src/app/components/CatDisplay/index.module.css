.catDisplayDiv {
  cursor: pointer;
  width: 100%;
}