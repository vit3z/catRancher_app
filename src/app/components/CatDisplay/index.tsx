import Image from "next/image";
import { Typography } from "@mui/material";
import { CatDisplayProps } from "@/consts/interfaces";
import styles from "./index.module.css";

const CatDisplay: React.FC<CatDisplayProps> = ({
  cat,
  catHeight,
  catWidth,
}) => {
  return (
    <div
      className={styles.catDisplayDiv}
    >
      <Image
        src={`https://static.quantcast.com/catrancher/${cat}.png`}
        width={catWidth}
        height={catHeight}
        alt={`Image of Cat with ID: ${cat}`}
      />
      {/* In theory, I didn't need to add this toUpperCase function, however, I found it easier to read the CatId with it */}
      <Typography variant="subtitle1">{cat.toUpperCase()}</Typography>
    </div>
  );
};

export default CatDisplay;
