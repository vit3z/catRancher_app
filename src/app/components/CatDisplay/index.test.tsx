import React from 'react';
import renderer from 'react-test-renderer';
import CatDisplay from './index';

jest.mock('next/image', () => 'img');
jest.mock('@mui/material', () => ({
  ...jest.requireActual('@mui/material'),
  Typography: jest.fn(({ variant, children }) => (
    <div className={`mock-typography ${variant}`}>{children}</div>
  )),
}));

describe('CatDisplay', () => {
  it('renders correctly', () => {
    const props = {
      cat: '1ttg',
      catHeight: 100,
      catWidth: 100,
      addCatToTentativeClowder: jest.fn(),
    };
    const component = renderer.create(<CatDisplay {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
