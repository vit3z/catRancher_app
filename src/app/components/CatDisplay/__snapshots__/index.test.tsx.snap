// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatDisplay renders correctly 1`] = `
<div
  className="catDisplayDiv"
>
  <img
    alt="Image of Cat with ID: 1ttg"
    height={100}
    src="https://static.quantcast.com/catrancher/1ttg.png"
    width={100}
  />
  <div
    className="mock-typography subtitle1"
  >
    1TTG
  </div>
</div>
`;
